libbpfgo/libbpfgo.go

```diff
@@ -20,16 +20,28 @@ package libbpfgo
 extern void perfCallback(void *ctx, int cpu, void *data, __u32 size);
 extern void perfLostCallback(void *ctx, int cpu, __u64 cnt);
 
+extern int ringbufferCallback(void *ctx, void *data, size_t size);
+
 int libbpf_print_fn(enum libbpf_print_level level,
-			   const char *format, va_list args)
+               const char *format, va_list args)
 {
-	if (level != LIBBPF_WARN)
-		return 0;
-	return vfprintf(stderr, format, args);
+    if (level != LIBBPF_WARN)
+        return 0;
+    return vfprintf(stderr, format, args);
 }
 
 void set_print_fn() {
-	libbpf_set_print(libbpf_print_fn);
+    libbpf_set_print(libbpf_print_fn);
+}
+
+struct ring_buffer * init_ring_buf(int map_fd) {
+    struct ring_buffer *rb = NULL;
+    rb = ring_buffer__new(map_fd, ringbufferCallback, NULL, NULL);
```

Comment:
Just to be consistent with the perf buffer init function, and to avoid passing `C.ringBufferCallback`. I can get rid of it if you feel strongly about it.

Reply:
I think it is better to avoid unnecessary wrappers and use go where possible to make code more readable.
Not a must though